### ISSUE TITLE
failed to resolve: use of undeclared crate or module `os`

### ISSUE THREAD

**Issue by PrisionMike** [johnstonskj/rust-upnp#2]
I added upnp to my toml file. Here's the full output after I ran a blank main with just use upnp_rs::sspd:

```
...
Compiling pnet_transport v0.25.0
   Compiling upnp-rs v0.1.0
error[E0433]: failed to resolve: use of undeclared crate or module `os`
  --> C:\Users\sush2\.cargo\registry\src\github.com-1ecc6299db9ec823\upnp-rs-0.1.0\src\utils\user_agent.rs:20:19
   |
20 |             name: os::system_name(),
   |                   ^^ use of undeclared crate or module `os`

error[E0433]: failed to resolve: use of undeclared crate or module `os`
  --> C:\Users\sush2\.cargo\registry\src\github.com-1ecc6299db9ec823\upnp-rs-0.1.0\src\utils\user_agent.rs:21:22
   |
21 |             version: os::system_version(),
   |                      ^^ use of undeclared crate or module `os`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0433`.
error: could not compile `upnp-rs`

```

**Comment by PrisionMike** [comment]
I tried addding `use std::os;` to the `user_agent.rs` file. It says `cannot find function system_name() in module os`

**Comment by PrisionMike** [comment]
Has the crate been tested on a Windows system? I am getting a LINK 1181 error after making some benign (aforementioned) modifications:

```
advapi32.lib" "ws2_32.lib" "userenv.lib" "msvcrt.lib"
  = note: LINK : fatal error LNK1181: cannot open input file 'Packet.lib'
```

**Comment by johnstonskj** [comment]
I have not tested on Windows unfortunately, I only have occasional access
to a stable Windows build machine.

**Comment by vaskoa** [comment]
Uh. I got the same problem when playing around on Windows.

While bypassing the non existent **os** module on Windows was trivial, afterwards, when trying to build the command line tool, I also got stuck on the same linking error:
```
 = note: LINK : fatal error LNK1181: cannot open input file 'Packet.lib'
```

It seems like this is a know problem of **pnet** and [they even address it in their readme](https://github.com/libpnet/libpnet#windows). And it also seems like other are [having the exact same problem](https://github.com/libpnet/libpnet/issues/332).

To be honest, installing **WinPcap | npcap** seemed like overkill for my use case, so I looked into the code so see what **pnet** was used for. To my surprise it is only to enumerate the available network interfaces, so I just removed the enumeration, forcing the tool to bind to address "0.0.0.0", which is actually preferable for me. I even managed to get some SSDP answers with it.

Hope this helps.
